pkg/config/config.go

```diff
@@ -1383,14 +1384,13 @@ func LoadProxyFromEnv(config Config) {
 	}
 }
 
-// Load reads configs files and initializes the config module
-func Load() (*Warnings, error) {
-	return LoadDatadogCustom(Datadog, "datadog.yaml", true, SystemProbe.GetEnvVars())
-}
-
 // LoadWithoutSecret reads configs files, initializes the config module without decrypting any secrets
 func LoadWithoutSecret() (*Warnings, error) {
-	return LoadDatadogCustom(Datadog, "datadog.yaml", false, SystemProbe.GetEnvVars())
+	return LoadDatadogCustom(Datadog, "datadog.yaml", nil, SystemProbe.GetEnvVars())
+}
+
+func LoadResolveSecret(secretResolver secrets.Component) (*Warnings, error) {
```

Comment:
To keep in line with the method above:
```suggestion
func LoadWithSecret(secretResolver secrets.Component) (*Warnings, error) {
```